examples/C++/DDS/BasicConfigurationExample/arg_configuration.h

```diff
@@ -0,0 +1,194 @@
+                }
+                return option::ARG_ILLEGAL;
+            }
+            return option::ARG_OK;
+        }
+        if (msg)
+        {
+            print_error("Option '", option, "' requires a string argument\n");
+        }
+        return option::ARG_ILLEGAL;
+    }
+
+};
+
+enum optionIndex
+{
+    UNKNOWN_OPT,
+    HELP,
+    TOPIC,
+    WAIT,
+    SAMPLES,
+    INTERVAL,
+    ASYNC,
+    DOMAIN_ID,
+    TRANSPORT,
+    RELIABLE,
+    TRANSIENT_LOCAL
+};
+
+const option::Descriptor usage[] = {
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,
+      "Usage: BasicConfigurationExample <publisher|subscriber>\n\nGeneral options:" },
+    { HELP,    0, "h", "help",               Arg::None,      "  -h \t--help  \tProduce help message." },
+
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nPublisher options:"},
+    { TOPIC, 0, "t", "topic",                  Arg::String,
+      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: HelloWorldTopic)." },
+    { DOMAIN_ID, 0, "d", "domain",                Arg::Numeric,
+      "  -d <id> \t--domain=<id>  \tDDS domain ID (Default: 0)." },
+    { WAIT, 0, "w", "wait",                 Arg::Numeric,
+      "  -w <num> \t--wait=<num> \tNumber of matched subscribers required to publish"
+      " (Default: 0 => does not wait)." },
+    { SAMPLES, 0, "s", "samples",              Arg::Numeric,
+      "  -s <num> \t--samples=<num>  \tNumber of samples to send (Default: 0 => infinite samples)." },
+    { INTERVAL, 0, "i", "interval",            Arg::Numeric,
+      "  -i <num> \t--interval=<num>  \tTime between samples in milliseconds (Default: 100)." },
+    { ASYNC, 0, "a", "async",               Arg::None,
+      "  -a \t--async \tAsynchronous publish mode (synchronous by default)." },
+    { TRANSPORT, 0, "", "transport",        Arg::Transport,
+      "  \t--transport=<shm|udp|udpv6> \tUse only data-sharing, shared-memory, UDPv4, UDPv6 transport."
+      "If not set, use Fast DDS default transports (depending on the scenario it will use the most efficient one)." },
+
+    { UNKNOWN_OPT, 0, "", "",                Arg::None,      "\nSubscriber options:"},
+    { TOPIC, 0, "t", "topic",                  Arg::String,
+      "  -t <topic_name> \t--topic=<topic_name>  \tTopic name (Default: HelloWorldTopic)." },
+    { DOMAIN_ID, 0, "d", "domain",                Arg::Numeric,
+      "  -d <id> \t--domain=<id>  \tDDS domain ID (Default: 0)." },
+    { SAMPLES, 0, "s", "samples",              Arg::Numeric,
+      "  -s <num> \t--samples=<num>  \tNumber of samples to wait for (Default: 0 => infinite samples)." },
+    { TRANSPORT, 0, "", "transport",        Arg::Transport,
+      "  \t--transport=<shm|udp|udpv6> \tUse only data-sharing, shared-memory, UDPv4, UDPv6 transport."
+      "If not set, use Fast DDS default transports (depending on the scenario it will use the most efficient one)." },
```

Comment:
I think it would be good to explicitly write which is the order of priority, including the words "data sharing delivery mechanism", since this is an example mainly for beginners.

Reply:
:+1: 
(Those words sound exactly what a beginner would like to know :laughing: )